src/algebra/ordered_ring.lean

```diff
@@ -605,6 +637,22 @@ lemma mul_pos_of_neg_of_neg {a b : α} (ha : a < 0) (hb : b < 0) : 0 < a * b :=
 have 0 * b < a * b, from mul_lt_mul_of_neg_right ha hb,
 by rwa zero_mul at this
 
+/-- Pullback an `ordered_ring` under an injective map. -/
+def function.injective.ordered_ring {β : Type*}
```

Comment:
Can you do `ordered_comm_ring` too? It should just be `..`ing two other defs

Reply:
I thought so as well, but I seem to have to prove these fields:

```lean
{ zero_mul := _,
  mul_zero := _,
  add_left_cancel := _,
  add_right_cancel := _,
  le_of_add_le_add_left := _,
  mul_lt_mul_of_pos_left := _,
  mul_lt_mul_of_pos_right := _,
  ..hf.ordered_ring f zero one add mul neg,
  ..hf.comm_ring f zero one add mul neg }
```

Reply:
Got it: I was missing `..hf.ordered_comm_semiring f zero one add mul`!